### PR TITLE
Apply various small BPF refactors

### PR DESCRIPTION
**Pull request by codingllama**
Pull a few BPF refactors and Vagrantile from #31900.

No behavioral changes.

### CODE REVIEW

**Review by zmb3 — APPROVED** [review]

**Review by rosstimothy — APPROVED** [review]

File: Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,78 @@
+# -*- mode: ruby -*-
```
Comment (jakule) [inline Vagrantfile:1]: If this is BPF testing VM, IMO should be in `bpf` directory. Keeping it in `/` suggests that this is the main Vagrant that should be used to run Teleport.

File: lib/bpf/common_test.go [hunk 2]
```diff
@@ -17,145 +17,62 @@ limitations under the License.
...
-		require.Equal(t, *tt.outConfig.DiskBufferSize, *tt.inConfig.DiskBufferSize)
-		require.Equal(t, *tt.outConfig.NetworkBufferSize, *tt.inConfig.NetworkBufferSize)
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			err := test.got.CheckAndSetDefaults()
+			require.NoError(t, err, "CheckAndSetDefaults errored")
+
+			if diff := cmp.Diff(test.want, test.got); diff != "" {
+				t.Errorf("CheckAndSetDefaults mismatch (-want +got)\n%s", diff)
+			}
```
Comment (jakule) [inline lib/bpf/common_test.go:75]: Let's be consistent
```suggestion
			require.Empty(t, cmp.Diff(test.want, test.got), "CheckAndSetDefaults mismatch (-want +got)\n%s", diff)
			
```

File: Vagrantfile (hunk 1 above)
Comment (codingllama) [reply Vagrantfile:1]: Done.

File: lib/bpf/common_test.go (hunk 2 above)
Comment (codingllama) [reply lib/bpf/common_test.go:75]: This just makes the error message worse to read, imo.

**Pull request merged**